subsys/bluetooth/host/gatt.c

```diff
@@ -5831,11 +5831,12 @@ static int db_hash_commit(void)
 	 * loaded from flash.
 	 */
 	if (IS_ENABLED(CONFIG_BT_LONG_WQ)) {
-		return bt_long_wq_reschedule(&db_hash.work, K_NO_WAIT);
+		(void)bt_long_wq_reschedule(&db_hash.work, K_NO_WAIT);
 	} else {
-		return k_work_reschedule(&db_hash.work, K_NO_WAIT);
+		(void)k_work_reschedule(&db_hash.work, K_NO_WAIT);
 	}
 
+	return 0;
```

Comment:
I'm confused. If the issue that they are returning a negative value, then this is working as intended IMO. The `commit` from `SETTINGS_STATIC_HANDLER_DEFINE(bt, "bt", NULL, set, commit, NULL);` works the same way. 

If the issue is that they are returning a positive value, then that value should be parsed, and any positive values should be returned as 0:

The proposed solution isn't good: If we really (really) do not want to return anything but `0` in this function, then we should at the very least log the return values in case of error with something like `BT_ERR`.

Reply:
These work API functions may return positive value which is improperly considered as error by settings. Improved error handling as requested (I think error would cannot happen with current implementation, but additional check would not hurt :) ).